### ISSUE TITLE
Globally WSL support

### ISSUE THREAD

**Issue by andre2** [Microsoft/vscode#50795]
How to works with WSL in VSCode.
Plugins always require `PATH` variable i.e. Flow Language Support or VSCode node debugger.
Main problem is in setup default terminal for VSCode. Git and Debugger don't work with VSCode. Using useWSL option in debugger results timeout. Git returns as not installed. After setup `terminal.integrated.shellArgs.windows` or `terminal.integrated.shell.windows` VSCode use `PATH` variable from Windows, not from WSL. Only terminal use this setup. Is any option to set terminal globaly for all addons?

- VSCode Version: 1.23.1
- OS Version: Windows 10 Pro

Steps to Reproduce:

1. User Settings:
WSL option examples here: https://docs.microsoft.com/en-us/windows/wsl/interop
```json
{
  "terminal.integrated.shellArgs.windows": ["wsl"],
}
```

Does this issue occur when all extensions are disabled?: Yes

**Comment by TiemenSch** [comment]
It would be great if there would be a global switch that makes VS Code use the Linux settings and tries to run the Linux commands in bash. If it's missing Linux dependencies, it would be up to the user to install those. No one is forcing them to use WSL.

CppTools has figured out a workaround that works for them using pipetransport:
https://github.com/Microsoft/vscode-cpptools/blob/master/Documentation/Debugger/gdb/Windows%20Subsystem%20for%20Linux.md

Perhaps a rudimentary implementation could be launch.json targets only.

**Comment by michaellee8** [comment]
Any update on this one? It will be wonderful if i may finally develop linux c programs with windows.

**Comment by msvaillant** [comment]
Now it seems clear that more and more people would come and ask for a way to work wsl + vscode, because this toolchain has immersive perspectives for all windows developers. Even to stop duplicating wsl and windows installs and developers stop making wsl support for each vscode extension.

So it would be nice to hear some feedback and roadmap on implementation of "Global wsl support" and for developers to upvote the comment for vscode-team to see we are interested in.

**Comment by xcdr** [comment]
This is also very important feature for me, without this I cannot switch to Windows.

**Comment by zenion** [comment]
i feel the same way... we really hope the vscode core team will embrace this as having to work in both environments and install everything redundantly to do basic vscode things is very problematic.

**Comment by phpcitizen** [comment]
+1

**Comment by RobertDeRose** [comment]
I was going to open a new issue for this, but this seems to address my basic idea. It would be great if Code just switched the nodejs platform to operate as if it was running on Linux when the user has set their terminal to WSL. This way things like child_processes and paths would all resolve as though they were on a Linux system. This would reduce the complexity for both users and extension developers. 

If I only had to worry about POSIX and Windows and not all the combination in between because of WSL, it would be a huge win as I imagine all extensions that just haven’t had time to add support or those that haven’t been well tested that just don’t work right. 

So in summary, when terminal is set to WSL, code treats underlaying OS as linux.

**Comment by mnxoid** [comment]
+1
This would be a great feature

**Comment by TiemenSch** [comment]
**STOP commenting +1's!** Use the emoji's on the top post.

**Comment by chrants** [comment]
This would provide excellent help for me working as the only person developing on Windows in a company of 95% Mac and 4% Linux developers. I maybe could convince more to join me if there was excellent out-of-the-box support of things like WSL.

**Comment by mutee88** [comment]
I have the exact problem. There are python libraries that work well for me on the linux subsystem but I need to do the code editing on VS code.  VS code prompts me to select a python environment but ofcourse the environment is not visible as it's in the Linux subsystem. It would be nice if we had such support.

**Comment by Perpetua-Kabute** [comment]
Any developments on this?

**Comment by SuperRoach** [comment]
I've ran into this as well, and would prefer to not need to run node setups side by side between windows and linux to have linting and other support

**Comment by Tyriar** [comment]
We just announced remote development with VS Code, check out the blog post for details https://code.visualstudio.com/blogs/2019/05/02/remote-development
